code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -175,20 +180,26 @@
 	if(!istype(exposed_turf))
 		return
 
-	var/cool_temp = cooling_temperature
+	// Water transfer to turf.
 	if(reac_volume >= 5)
 		exposed_turf.MakeSlippery(TURF_WET_WATER, 10 SECONDS, min(reac_volume*1.5 SECONDS, 60 SECONDS))
 
+	// Water application to slimes.
 	for(var/mob/living/simple_animal/slime/exposed_slime in exposed_turf)
 		exposed_slime.apply_water()
 
+	// Heat exchange with the air.
+	var/datum/gas_mixture/air = exposed_turf.return_air()
+	if(!istype(air, /datum/gas_mixture/immutable))
+		var/temperature_delta = holder.chem_temp - air.temperature
+		var/air_heat_capacity = air.heat_capacity()
+		var/conduction_energy = CALCULATE_CONDUCTION_ENERGY(temperature_delta, air_heat_capacity, atmos_heat_exchange_specific_heat * reac_volume)
+		air.temperature = air.temperature + conduction_energy / air_heat_capacity
```

Comment:
how useful are these comments? they should be pretty obvious just from reading the code, right?
I think it's better to describe the stuff that people can't intuit, at least like this.

don't get me wrong documenting at all is based but it should exist to explain stuff you can't just read (I realize that depends on the reader)

Reply:
Oh also, mark the turf for activation 
even better use temperature_share(), so you don't need to do a typecheck for immutable like this. smells wrong

Reply:
Weird phenomena occurs when I use temperature_share(), probably because it is using archived temperature.

Reply:
hhhhhhhhhhhhhhhhhhhhhhhhhhhhhhh

Reply:
Why does temperature_share() use the archived_temperature for temperature delta anyways?

Reply:
it's primarially used in share() and supercon, and optimized for that usecase. Might be good to do some spring cleaning on it honestly, so we can have a macro and a proc one for tile processing and one for this sort of thing. IDK immutable mixes are annoying